### PR TITLE
Always display NA as unquoted in the Variables pane

### PR DESCRIPTION
**Pull request by yutannihilation**
This pull request addresses https://github.com/posit-dev/positron/issues/3810

Since `format()` and `format_one()` belongs to `Vector`, I think `format_with_string_options()` should also belong to `Vector` and be called there. But, I'm not sure if it's a good idea to include `FormattedVectorCharacterOptions` in the signature of all types of `Vector`. Considering https://github.com/posit-dev/positron/issues/2860 will introduce a facility to handle this kind of distinction easier, I'm not sure if this is worth, but I share this code just in case this helps.

### CODE REVIEW

**Review by lionel- — COMMENTED** [review]
I don't mind having it in the signature, the added verbosity has the advantage that callers get reminded of formatting options. @dfalbel Since you worked on formatting, could you review this PR please?

File: crates/harp/src/vector/mod.rs [hunk 1]
```diff
@@ -100,12 +101,32 @@ pub trait Vector {
...
+    fn format_one(
+        &self,
+        x: Self::Type,
+        options: Option<&FormattedVectorCharacterOptions>,
+    ) -> String;
+
+    fn format_elt_unchecked(
+        &self,
+        index: isize,
+        options: Option<&FormattedVectorCharacterOptions>,
```
Comment (dfalbel) [inline crates/harp/src/vector/mod.rs:113]: I think we should move the definition of `FormattedVectorCharacterOptions` into this file. 
We probably also want to change it's name to something like `FormatOptions` as it's no longer specific to character vectors.

File: crates/harp/src/vector/mod.rs [hunk 2]
```diff
@@ -100,12 +101,32 @@ pub trait Vector {
...
+    ) -> String {
         match self.get_unchecked(index) {
-            Some(x) => self.format_one(x),
+            Some(x) => self.format_one(x, options),
             None => String::from("NA"),
         }
     }
+
+    fn format_with_string_options(
```
Comment (dfalbel) [inline crates/harp/src/vector/mod.rs:121]: I think this should be renamed to `format_with_options`.

**Review by dfalbel — COMMENTED** [review]
These options were introduced in order to be able to control the formatting for the data explorer, but we no longer use the `FormattedVector` interface (see https://github.com/posit-dev/ark/pull/382) because this interface wasn't flexible enough to support both use cases. AFAICT `FormattedVector` is currently only used by the Variables pane, so we can make it specific for it.

Maybe a better approach is to simply remove those and we can add it back if we need it at some point. 

We should also add a test for the `NA` formatting here: https://github.com/posit-dev/ark/blob/0071f7a90ce035066883cfdab45248c4dd6fbdf9/crates/harp/src/vector/formatted_vector.rs#L242

**Comment by yutannihilation** [comment]
Thanks for the review! Renamed the struct and add a test.

Also, thanks for the context. So, maybe it needs some more cleanup? I want to keep this pull request small for now (mainly because I don't know well about the implementations), but I can address in a separate pull request.

File: crates/harp/src/vector/mod.rs [hunk 3]
```diff
@@ -100,12 +107,20 @@ pub trait Vector {
...
-    fn format_elt_unchecked(&self, index: isize) -> String {
+    fn format_elt_unchecked(&self, index: isize, options: Option<&FormatOptions>) -> String {
         match self.get_unchecked(index) {
-            Some(x) => self.format_one(x),
+            Some(x) => self.format_one(x, options),
             None => String::from("NA"),
         }
     }
+
+    fn format_with_options(&self, value: String, options: &FormatOptions) -> String {
```
Comment (dfalbel) [inline crates/harp/src/vector/mod.rs:119]: If this only affects the character vectors, this should be moved to the `character_vector.rs` file.

File: crates/harp/src/vector/mod.rs [hunk 4]
```diff
@@ -37,6 +37,13 @@ pub use raw_vector::RawVector;
 pub mod formatted_vector;
 pub mod names;
 
+// Formatting options for character vectors
+pub struct FormatOptions {
+    // Wether to quote the strings or not (defaults to `true`)
+    // If `true`, elements will be quoted during format so, eg: c("a", "b") becomes ("\"a\"", "\"b\"") in Rust
```
Comment (dfalbel) [inline crates/harp/src/vector/mod.rs:43]: We should document that the `quote` option only affects character vectors.

**Review by dfalbel — APPROVED** [review]
Thanks @yutannihilation for the PR! I have just added a few more comments, but this looks great!

**Comment by yutannihilation** [comment]
Thanks! I inlined `format_with_options()` and added a note.

**Pull request merged**